module #2. REST API Basics/rest-api-basic/core/src/main/java/com/epam/esm/repository/TagRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.model.Tag;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface TagRepository {
+    boolean isExists(Tag tag);
+    Long save(Tag tag);
+    Optional<Tag> findById(Long id);
+    Optional<Tag> findByName(String name);
+    Optional<List<Tag>> findAll();
+    void deleteById(Long id);
```

Comment:
Please, do the same for TagMappingServiceImpl as it is described in GiftCertificateRepository. TagMappingServiceImpl should implements CrudRepository<...> with required generic arguments